core/src/main/scala/scalaz/Maybe.scala

```diff
@@ -390,6 +392,11 @@ sealed abstract class MaybeInstances extends MaybeInstances0 {
 
       override def filter[A](fa: Maybe[A])(f: A => Boolean): Maybe[A] =
         fa.filter(f)
+
+      override def alt[A](a1: => Maybe[A], a2: => Maybe[A]): Maybe[A] = altly2(a1, a2) {
```

Comment:
and don't forget to implement `altly2` or you'll get an infinite loop too

Reply:
for `Maybe`, it might make sense to implement `ap` and `alt` AND `apply2` / `altly2` for performance reasons. Only one set is needed, but I can imagine `Maybe` being used in perf critical code a lot.

Reply:
I've implemented altly2 and apply2.